Assets/_Functionalities/Wms/Scripts/ImageProjectionLayer.cs

```diff
@@ -155,6 +162,80 @@ protected virtual Tile CreateNewTile(Vector2Int tileKey)
             return tile;
         }
 
+        public void RefreshTiles()
+        {
+            //is the update already running cancel it
+            if (visibleTilesDirty && updateTilesRoutine != null)
+            {
+                queuedChanges.Clear();
+                StopCoroutine(updateTilesRoutine);
+            }
+
+            lastUpdatedTimeStamp = Time.time;
+            visibleTilesDirty = true;
+            updateTilesRoutine = StartCoroutine(UpdateVisibleTiles());
+        }
+
+        private IEnumerator UpdateVisibleTiles()
+        {
+            //get current tiles
+            foreach (KeyValuePair<Vector2Int, Tile> tile in tiles)
+            {
+                if (tile.Value == null || tile.Value.gameObject == null)
+                    continue;
+
+                if (tile.Value.runningCoroutine != null)
+                    StopCoroutine(tile.Value.runningCoroutine);
+
+                OnPreUpdateTile(tile.Value);
+
+                TileChange tileChange = new TileChange();
+                tileChange.X = tile.Key.x;
+                tileChange.Y = tile.Key.y;
+                queuedChanges.Add(tileChange);
+            }
+
+            if (!isEnabled)
+            {
+                queuedChanges.Clear();
+                yield break;
+            }
+
+            bool ready = true;
+            while (queuedChanges.Count > 0)
+            {
+                //lets wait half a second in case a slider is moving
+                if (Time.time - lastUpdatedTimeStamp > lastUpdatedInterval && ready)
+                {
+                    ready = false;
+                    TileChange next = queuedChanges[0];
+                    queuedChanges.RemoveAt(0);
+                    Vector2Int key = new Vector2Int(next.X, next.Y);
+                    if (tiles.ContainsKey(key))
```

Comment:
Als deze een aparte methode had geweest, dan had je deze if kunnen inverten en zou de code leesbaarder zijn

Reply:
voor nu zo gelaten omdat het een stijl comment is